File: examples/chatter/listener.cc
#include <example_msgs/Example.h>
#include <ros/ros.h>

void ChatterCallback(const example_msgs::Example& msg) {
  ROS_INFO_STREAM("Received: " << msg.message);
}

int main(int argc, char** argv) {
  ros::init(argc, argv, "listener");

  ros::NodeHandle n;
  ros::Subscriber chatter_sub = n.subscribe("chatter", 1000, ChatterCallback);

  ros::spin();

  return 0;
}

File: examples/chatter/listener.py
import example_msgs.msg
import rospy


def callback(msg):
    rospy.loginfo(f'Python listener received: {msg.message}')


def main():
    rospy.init_node('listener', anonymous=True)
    rospy.Subscriber('chatter', example_msgs.msg.Example, callback)
    rospy.spin()


if __name__ == '__main__':
    main()
